### ISSUE TITLE
[BUG] authenticate to private git repository hosted on Azure DevOps using using HTTP Basic Auth with PAT fails when using it as Rancher Chart Repository since rancher 2.10.x

### ISSUE THREAD

**Issue by aDisplayName** [rancher/rancher#48388]
**Rancher Server Setup**
- Rancher version: 2.10.0
- Installation option: Helm

**Information about the Cluster**
- Kubernetes version:
- Cluster Type (Local/Downstream):
   Downstream, Imported, k3s 1.28.13

**User Information**
- What is the role of the user logged in?
  Admin/Cluster Owner

**Describe the bug**
For a Rancher Chart repository created using "Git Repository" option, if the git repository is a private git repository hosted on https://dev.azure.com, the git clone will fail starting rancher 2.10.x

The git clone did not fail in 2.8.4

**To Reproduce**
1. In Cluster Dashboard &rarr; Apps &rarr; Repositories, create a new Chart Repository:
   * Select `Git repository containing Helm chart or cluster template definitions`
   * For index URL: such as `https://dev.azure.com/sample-org/project/_git/repo-name`
   * Create a new HTTP Basic Auth using "azure" as username, and Azure PAT as password.
1. Save

**Result**
The repository will then go to Error status, with following messages:
```
Head failure: git clone --depth=1 -n -- https://azure@dev.azure.com/sample-org/project/_git/repo-name management-state/git-repo/git-azure/fcbbf3098e5a1db4837249705873110484603daf98bdf83db1d668183aab1c4f error: exit status 128, detail: Cloning into 'management-state/git-repo/git-azure/fcbbf3098e5a1db4837249705873110484603daf98bdf83db1d668183aab1c4f'... fatal: unable to update url base from redirection: asked for: https://azure@dev.azure.com/sample-org/project/_git/repo-name/info/refs?service=git-upload-pack redirect: https://spsprodweu2.vssps.visualstudio.com/_signin?realm=dev.azure.com&reply_to=xxxxxxxxxxxxxxxxxxx . Will retry after 18m54s
```
**Expected Result**
The Rancher chart repository based on Azure Devops Private git repository should be initialized successfully

**Screenshots**
![image](https://github.com/user-attachments/assets/ad06a6b7-eb3c-4cec-804d-9535ac25cc75)

**Additional context**
The same setting works well in Rancher 2.8.4
The rancher server is then upgraded to 2.10.0

Tried to create a new repository in Rancher 2.10.0 resulted in the same error, so we don't think it is related to the upgrade.

**Comment by ymor** [comment]
Same issue here but when the repo is hosted by Bitbucket

**Comment by krosk** [comment]
> Same issue here but when the repo is hosted by Bitbucket.
> 
> Update: SSH is working as expected it seems to be just an issue with basic auth

Do you mind sharing which URL format you used to point to a repo via SSH? I get various flavors of "invalid git URL, only http(s) or ssh:// supported"

**Comment by krosk** [comment]
Thanks for confirming.

For those stuck in a similar situation with ADO, I could not use the format `git@ssh.dev.azure.com:v3/organisation/project/repo` because the organization name had hyphens (https://github.com/rancher/rancher/issues/45610). 

That post did suggest a workaround which worked. So in rancher, I specify `ssh://git@ssh.dev.azure.com:22/v3/organisation/project/repo`

**Comment by dncman** [comment]
I could observe the same issue with Gitlab. For me using HTTPS + Basic Auth worked when using `https://<username>:<password>@my.git.server.com/repo/fancyproject.git` with using `None` as Authentification in Rancher.

Like this:
![Image](https://github.com/user-attachments/assets/534cc844-bc1e-4a15-800f-e76bbceed80e)

Not ideal and quite hacky but it's an alternative for everyone who can't or don't want to use SSH. 

I hope the Rancher team fixes it in the next minor release. I think there is an issue in applying/using the password when doing the git clone. It seems like it just uses the username but not the password from the secret.

**Comment by deepakpunia-suse** [comment]
Below is the bug verification details. 

Details
rancher: v2.10-head
RKE2 : v1.30.10-rc1+rke2r1

1. In the Rancher Cluster Explorer, navigate to Apps -> Repositories.
2. Click "Add Repository."
3. Select "Git repository containing Helm chart or cluster template definitions."
4. In the "Index URL" field, enter the URL of your Git repository (e.g., `https://<username>@bitbucket.org/chartrepo/chartsrepo.git>`).  
5. Create a new "HTTP Basic Auth" entry.  Enter a username and password. 
6. Save the repository.


The repository should be added successfully and have an "Active" status and upgrade also working fine. 

![Image](https://github.com/user-attachments/assets/1408e886-3cbd-4b17-abc1-f7ec4fdb9a9b)

![Image](https://github.com/user-attachments/assets/cc44dc54-efc7-4a39-85c4-fa86c3712f7c)

**Comment by johnwc** [comment]
> Below is the bug verification details.
> 
> Details rancher: v2.10-head RKE2 : v1.30.10-rc1+rke2r1
> 
> 1. In the Rancher Cluster Explorer, navigate to Apps -> Repositories.
> 2. Click "Add Repository."
> 3. Select "Git repository containing Helm chart or cluster template definitions."
> 4. In the "Index URL" field, enter the URL of your Git repository (e.g., `https://<username>@bitbucket.org/chartrepo/chartsrepo.git>`).
> 5. Create a new "HTTP Basic Auth" entry.  Enter a username and password.
> 6. Save the repository.
> 
> The repository should be added successfully and have an "Active" status and upgrade also working fine.
> 
> ![Image](https://github.com/user-attachments/assets/1408e886-3cbd-4b17-abc1-f7ec4fdb9a9b)
> 
> ![Image](https://github.com/user-attachments/assets/cc44dc54-efc7-4a39-85c4-fa86c3712f7c)

@deepakpunia-suse you tested against bitbucket, but the issue was raised for failing to connect to Azure DevOps.

**Comment by rohitsuse** [comment]
@johnwc I will look into this.

**Comment by vasiok3000** [comment]
rancher_server = "2.10.3" rke2_version   = "v1.31.7+rke2r1" In this versions was solved for me, I'm using azure devops server 2022.

**Comment by johnwc** [comment]
We are using v2.11.2 with v1.32.5+rke2r1, it still fails with http basic auth.

@rohitsakala can the issue be reopened until you validate it?

**Comment by johnwc** [comment]
@rohitsakala We are getting nothing but failures now, nothing we do makes it work. Just same error. This is adding the repo in fleet for cluster CI/CD templates. If I download the same version of fleet and run it with the exact same command, it works. The init container is using `rancher/fleet:v0.12.3` image.

```
Error (time="2025-06-25T19:29:21Z" level=fatal msg="failed to clone repo from branch repo=\"ssh://git@ssh.dev.azure.com:22/v3/OurOrg/OurProject/Clusters\" branch=\"master\" revision=\"\" path=\"/workspace\": ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain" )
```

**Comment by rohitsuse** [comment]
@johnwc Can you specify if you are using `Apps & Marketplace` to add the helm repository ? If possible can you provide a screenshot of the UI where you are adding this Helm Repository.

**Comment by johnwc** [comment]
@rohitsakala We are adding it via the fleet interface. It's the same exact error, we changed it to ssh like suggested above it and started working a few weeks back. But now it fails.

![Image](https://github.com/user-attachments/assets/e6bcb9e8-6692-4ce0-8ace-5ff8466668c7)

**Comment by rohitsuse** [comment]
@johnwc Can you please try from the `Apps & Marketplace` interface ?

**Comment by johnwc** [comment]
@rohitsakala it seems to still be working in the Apps & Marketplace interface, just not the fleet Git Repos interface.

**Comment by rohitsuse** [comment]
thank you for confirming that to me @johnwc . I would like to close this issue since this issue was created for Apps & Marketplace interface. 

The issue is with fleet. Please create an issue with fleet https://github.com/rancher/fleet

**Comment by johnwc** [comment]
@rohitsakala no, there is two issues here, both with same cause/outcome. The http still does not work with azure devops private repos in Apps & Marketplace.

**Comment by rohitsakala** [comment]
\forward v2.13.0
